docs/DeveloperGuide.md

```diff
@@ -2,37 +2,731 @@
+
+Code Snippet
+```
+MemberList members = group.getMemberList();
+TransactionList transactions = group.getTransactionList();
+transactions.addTransaction(taskExpression, members);
+group.updateTransactionSolution();
+group.saveAllData();
+```
+<ins>Conclusion</ins>
+
+The Transaction class provides functionality for managing transactions between
+facilitates the lending of money and ensures data integrity by validating input and managing member interactions.
+
+The TransactionList class provides essential functionalities for managing a list of transactions.
+Its methods facilitate the addition, removal, editing, and retrieval of transactions, ensuring efficient management of transactions within a group.
+
+### PIN
+
+<ins> Overview </ins>
+
+The PINHandler class is responsible for managing the creation, loading, authentication, and resetting of a
+Personal Identification Number (PIN) used for authentication in the LongAh application. It uses SHA-256 hashing to
+securely store and compare PINs. The PINHandler class interacts with the StorageHandler class to save and load the PIN
+and authentication status.
+
+Note: PIN is enabled by default and needs to be set upon first startup.
+
+<ins>Implementation Details </ins>
+
+*Data Storage:*
+
+The PIN and authentication enabled status are saved in a file located at ./data/pin.txt.
+The file format is as follows:
+
+`hashed PIN`<br />
+`authenticationEnabled`<br />
+
+<ins> Class Structure </ins>
+
+The PINHandler class has the following static fields:
+
+
+- *PIN_FILE_PATH*: The path to the file where the PIN and authentication status are saved.
+
+- *savedPin*: The hashed PIN saved in the file.
+
+- *authenticationEnabled*: A boolean flag indicating whether authentication is enabled.
+
+<ins> Constructor </ins>
+
+The PINHandler constructor initializes the savedPin and authenticationEnabled fields by loading them from the file using
+the loadPinAndAuthenticationEnabled method.
+
+If the file does not exist or the savedPin is empty, it calls the createPin method to create a new PIN.
+
+<ins> Methods </ins>
+
+- *loadPinAndAuthenticationEnabled*: Loads the saved PIN and authentication enabled status from the file.
+
+- *savePinAndAuthenticationEnabled*: Saves the PIN and authentication enabled status to the file.
+
+- *getPinFilePath*: Returns the file path of the PIN file.
+
+- *createPin*: Prompts the user to create a new 6-digit PIN and hashes it before saving.
+
+- *authenticate*: Authenticates the user by comparing the entered PIN with the saved PIN.
+
+- *resetPin*: Resets the PIN for the user by prompting for the current PIN and creating a new PIN if the current
+PIN is correct.
+
+- *enablePin*: Enables authentication upon startup.
+
+- *disablePin*: Disables authentication upon startup.
+
+- *getSavedPin*: Returns the saved PIN.
+
+- *getAuthenticationStatus*: Returns the authentication status.
+
+<ins> Usage Example </ins>
+
+![pinhandler longah.png](diagrams%2Fpinhandler%20longah.png)
```

Comment:
Class LongAh calls new PINHandler(). This should be the label for the arrow from LongAh instead of Initialise. Also, as in this case the constructor of PINHandler() is called, the activation bar should be directly below the class name :PINHandler